packages/flutter/lib/src/material/carousel.dart

```diff
@@ -448,6 +452,40 @@ class CarouselView extends StatefulWidget {
   /// The child widgets for the carousel.
   final List<Widget> children;
 
+  /// A callback invoked when the leading item changes.
+  ///
+  /// The “leading” item is the one resolved as primary within the viewport.
+  /// Its definition depends on the carousel configuration:
+  ///
+  /// - For a standard [CarouselView], it is the item
+  ///   that becomes fully visible at the start of the viewport.
+  ///
+  /// - For a [CarouselView.weighted], it is the
+  ///   visible item that occupies the primary position, determined by
+  ///   the largest effective `weight` in the current layout pass.
```

Comment:
What if the largest weight is not unique? As in, the same largest weight is used on two items?

Reply:
It seems we call `widget.onIndexChanged?.call(currentLeadingIndex);` as soon as currentLeadingIndex is changed. If so, it is NOT the visible item that occupies the primary position, but just the leading item. We might need to correct the doc here.